core/src/main/java/org/infinispan/interceptors/CacheLoaderInterceptor.java

```diff
@@ -93,46 +87,22 @@ protected void injectDependencies(PersistenceManager clm, EntryFactory entryFact
 
    @Override
    public Object visitApplyDeltaCommand(InvocationContext ctx, ApplyDeltaCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
-         }
-      }
-      return invokeNextInterceptor(ctx, command);
+      return visitDataCommand(ctx, command);
    }
 
    @Override
    public Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
```

Comment:
Agree with @rvansa, but I think the Invalidate should not load anything from the CacheLoader. It should remove the entry from memory and from private CacheStore (if not L1 invalidate). Anyway, this is outside the scope of this PR :)